### PR TITLE
Wg 34 generate workout api call

### PR DESCRIPTION
**Pull request by chrismcanulty**
Perform API call to GET subset of exercise data based on user inputs stored in context state. Once retrieved, remove any duplicates based on exercise id and shuffle exercise order, then return sliced data based on user specified number of exercises. Render in unstyled flatlist

JIRA ticket: https://lhl-midterm-ecommerce.atlassian.net/browse/WG-34

Screenshot:

Screen Shot 2023-04-06 at 3 24 31 PM

### CODE REVIEW

File: src/context/User.Context.js [hunk 1]
```diff
@@ -53,6 +54,53 @@ const UserProvider = ({children}) => {
...
+
+    let tempEquipmentArr = [];
+    for (let i = 0; i < selectedEquipment.length; i++) {
+      const [filteredEquipmentTypes] = equipmentTypes.filter(
+        equipment => equipment.name === selectedEquipment[i],
+      );
+      tempEquipmentArr.push(filteredEquipmentTypes.id);
+    }
+    const equipmentIds = tempEquipmentArr.join(',');
+
```
Comment (wadesun18) [inline src/context/User.Context.js:76]: You’re doing the same thing twice right? Can you create a function here so you can just call it with different params

File: src/context/User.Context.js [hunk 2]
```diff
@@ -53,6 +54,53 @@ const UserProvider = ({children}) => {
...
+        return [...new Map(arr.map(item => [item[key], item])).values()];
+      }
+
+      function shuffleData(array) {
+        let i = array.length;
+        while (i--) {
+          const ri = Math.floor(Math.random() * i);
+          [array[i], array[ri]] = [array[ri], array[i]];
+        }
+        return array;
```
Comment (wadesun18) [inline src/context/User.Context.js:92]: Can these function live outside?

File: src/context/User.Context.js [hunk 3]
```diff
@@ -53,6 +54,53 @@ const UserProvider = ({children}) => {
...
+      }
+
+      const filteredData = getUniqueListBy(res.data.results, 'id');
+      const shuffledData = shuffleData(filteredData);
+
+      setExerciseData(shuffledData.slice(0, numberOfExercises));
```
Comment (wadesun18) [inline src/context/User.Context.js:98]: 👍

File: src/screens/HomeScreen.tsx [hunk 4]
```diff
@@ -86,6 +86,10 @@ export default function HomeScreen({navigation}: NativeStackHeaderProps) {
     }
   };
 
+  const setSelectedMusclesData = val => {
+    console.log(val);
+  };
+
```
Comment (wadesun18) [inline src/screens/HomeScreen.tsx:92]: ?

**Review by wadesun18 — CHANGES_REQUESTED** [review]

File: src/context/User.Context.js (hunk 2 above)
Comment (chrismcanulty) [reply src/context/User.Context.js:92]: Do you want me to move to a different file and import into this one?

**Pull request merged**